libqtile/window.py

```diff
@@ -1316,3 +1316,14 @@ def cmd_set_position(self, dx, dy, curx, cury):
                 self.group.layout.focused = index2
                 self.group.layoutAll()
                 break
+
+    def __getstate__(self):
+        d = {}
+        d['group'] = self.group.name
+        d['window'] = self.window.wid
+        return d
+
+    def __setstate__(self, state):
+        self.Wid = state['window']
```

Comment:
I purposefully made it `Wid` as it was an attribute name no one had , I was earlier using this to differentiate between a pickled window class and non-pickled one.

`self.gid` is an unused atttribute , I will remove it.